src/main/java/org/folio/dataexp/service/export/strategies/InstancesExportStrategy.java

```diff
@@ -4,26 +4,24 @@
 import org.folio.dataexp.domain.entity.JobExecutionExportFilesEntity;
 import org.folio.dataexp.domain.entity.JobExecutionExportFilesStatus;
 import org.folio.dataexp.repository.JobExecutionExportFilesEntityRepository;
-import org.folio.dataexp.service.export.storage.FolioS3ClientFactory;
+import org.folio.s3.client.FolioS3Client;
 import org.folio.s3.client.RemoteStorageWriter;
 import org.springframework.stereotype.Component;
 
 @Component
 @AllArgsConstructor
 public class InstancesExportStrategy implements ExportStrategy {
 
-  private final FolioS3ClientFactory folioS3ClientFactory;
   private final JobExecutionExportFilesEntityRepository jobExecutionExportFilesEntityRepository;
+  private final FolioS3Client s3Client;
 
   @Override
   public ExportStrategyStatistic saveMarcToRemoteStorage(JobExecutionExportFilesEntity exportFilesEntity) {
     var marc = "marc";
-    var s3Client = folioS3ClientFactory.getFolioS3Client();
     var remoteStorageWriter = new RemoteStorageWriter(exportFilesEntity.getFileLocation(),  8192, s3Client);
     remoteStorageWriter.write(marc);
     remoteStorageWriter.close();
```

Comment:
This can be done with try-with-resources.

Reply:
Updated thanks